cmake/CompileRocksDB.cmake

```diff
@@ -5,64 +5,54 @@ find_package(RocksDB 9.7.3)
 include(ExternalProject)
 
 set(RocksDB_CMAKE_ARGS
-  -DUSE_RTTI=1
-  -DPORTABLE=${PORTABLE_ROCKSDB}
-  -DFORCE_SSE42=${ROCKSDB_SSE42}
-  -DFORCE_AVX=${ROCKSDB_AVX}
-  -DFORCE_AVX2=${ROCKSDB_AVX2}
-  -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
-  -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
-  -DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}
-  -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
-  -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-  -DCMAKE_SHARED_LINKER_FLAGS=${CMAKE_SHARED_LINKER_FLAGS}
-  -DCMAKE_STATIC_LINKER_FLAGS=${CMAKE_STATIC_LINKER_FLAGS}
-  -DCMAKE_EXE_LINKER_FLAGS=${CMAKE_EXE_LINKER_FLAGS}
-  -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-  -DCMAKE_EXPORT_COMPILE_COMMANDS=${CMAKE_EXPORT_COMPILE_COMMANDS}
-  -DFAIL_ON_WARNINGS=OFF
-  -DWITH_GFLAGS=OFF
-  -DWITH_TESTS=OFF
-  -DWITH_TOOLS=${ROCKSDB_TOOLS}
-  -DWITH_CORE_TOOLS=${ROCKSDB_TOOLS}
-  -DWITH_BENCHMARK_TOOLS=OFF
-  -DWITH_BZ2=OFF
-  -DWITH_LZ4=ON
-  -DWITH_SNAPPY=OFF
-  -DWITH_ZLIB=OFF
-  -DWITH_ZSTD=OFF
-  -DWITH_LIBURING=${WITH_LIBURING}
-  -DWITH_TSAN=${USE_TSAN}
-  -DWITH_ASAN=${USE_ASAN}
-  -DWITH_UBSAN=${USE_UBSAN}
-  -DROCKSDB_BUILD_SHARED=OFF
-  -DCMAKE_POSITION_INDEPENDENT_CODE=True
-)
+    -DUSE_RTTI=1
+    -DPORTABLE=${PORTABLE_ROCKSDB}
+    -DFORCE_SSE42=${ROCKSDB_SSE42}
+    -DFORCE_AVX=${ROCKSDB_AVX}
+    -DFORCE_AVX2=${ROCKSDB_AVX2}
+    -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
+    -DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}
+    -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+    -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+    -DCMAKE_SHARED_LINKER_FLAGS=${CMAKE_SHARED_LINKER_FLAGS}
+    -DCMAKE_STATIC_LINKER_FLAGS=${CMAKE_STATIC_LINKER_FLAGS}
+    -DCMAKE_EXE_LINKER_FLAGS=${CMAKE_EXE_LINKER_FLAGS}
+    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+    -DCMAKE_EXPORT_COMPILE_COMMANDS=${CMAKE_EXPORT_COMPILE_COMMANDS}
+    -DFAIL_ON_WARNINGS=OFF
+    -DWITH_GFLAGS=OFF
+    -DWITH_TESTS=OFF
+    -DWITH_TOOLS=${ROCKSDB_TOOLS}
+    -DWITH_CORE_TOOLS=${ROCKSDB_TOOLS}
+    -DWITH_BENCHMARK_TOOLS=OFF
+    -DWITH_BZ2=OFF
+    -DWITH_LZ4=ON
+    -DWITH_SNAPPY=OFF
+    -DWITH_ZLIB=OFF
+    -DWITH_ZSTD=OFF
+    -DWITH_LIBURING=${WITH_LIBURING}
+    -DWITH_TSAN=${USE_TSAN}
+    -DWITH_ASAN=${USE_ASAN}
+    -DWITH_UBSAN=${USE_UBSAN}
+    -DROCKSDB_BUILD_SHARED=OFF
+    -DCMAKE_POSITION_INDEPENDENT_CODE=True)
 
 if(ROCKSDB_FOUND)
-  ExternalProject_Add(rocksdb
-    SOURCE_DIR "${RocksDB_ROOT}"
-    DOWNLOAD_COMMAND ""
-    CMAKE_ARGS ${RocksDB_CMAKE_ARGS}
-    BUILD_BYPRODUCTS <BINARY_DIR>/librocksdb.a
-    INSTALL_COMMAND ""
-  )
-
-  ExternalProject_Get_Property(rocksdb BINARY_DIR)
-  set(ROCKSDB_LIBRARIES
-      ${BINARY_DIR}/librocksdb.a)
+  set(ROCKSDB_LIBRARIES ${ROCKSDB_LIBRARY})
```

Comment:
did you test this both
* with rocksdb already installed
* without rocksdb already installed

Reply:
Tried on my devpods and works so far.